### PR TITLE
CI Fix: Prefix script-paths with ENVOY_SRCDIR

### PR DESCRIPTION
**Pull request by murray-stripe**
**Commit Message:**
```text
Update `do_ci.sh` to prefix paths to scripts with ENVOY_SRCDIR to
support running scripts outside of the Envoy root directory (for
example, for envoy-filter-example style setups)

Signed-off-by: John Murray <murray@stripe.com>
```
**Additional Description:** n/a
**Risk Level:** low
**Testing:** Tested manually in docker container. Will be covered by CI tests
**Docs Changes:** n/a
**Release Notes:** n/a
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by lizan — DISMISSED** [review]

**Review by phlax — CHANGES_REQUESTED** [review]
if we are going to do this (which i think is probably a good idea) can we catch all of them ?

there are quite a few more that use relative paths: 

eg: https://github.com/envoyproxy/envoy/blob/b66a5023a94c9fc607addab2c5fddc8eaa89b368/ci/do_ci.sh#L350

**Comment by murray-stripe** [comment]
Can definitely update all the relative paths. Will update in a bit.

**Review by phlax — APPROVED** [review]
lgtm, thanks @murray-stripe

**Comment by phlax** [comment]
@murray-stripe this has hit a problem because it assumes that `ENVOY_SRCDIR` is set

most of the ci jobs (not all) pass because `ENVOY_SRCDIR` is set in the ubuntu-build-image

personally i never use it - which also means this would break for some devs

a possible solution is to set it to `$PWD` if it has not already been set (which i think you need to do for this PR)

its highlighting the duplication/confusion of env vars however

**Comment by murray-stripe** [comment]
@phlax - I've updated to set `ENVOY_SRCDIR` and I've also updated `SRCDIR` to conditionally set. I've defaulted both to `$PWD`, which I believe is what we want in the default case. In general these are _very_ useful envvars for build setups similar to envoy-filter-example, although I would probably agree that they're not the most well documented. 

Happy to add a comment block to the script for these two variables if we want to inject some clarity into what they are and how they're used. Also don't want to clutter a script-file too much if these values are repeated in multiple scripts, so holding off for your guidance.

**Comment by phlax** [comment]
i guess im wondering if both are necessary - but even if not probs we dont want to just remove

feel free to add a `TODO` (for me if you wish) to say clarify or integrate these env vars

File: ci/do_ci.sh
Comment (phlax) [inline ci/do_ci.sh:15]: great thanks

**Comment by murray-stripe** [comment]
Adding `export`s to the top values to ensure that they are set properly in all scripts called from this one. Could also pass to individual script invocations as needed, but perhaps it's better to ensure all scripts can make the same set of assumptions?

**Comment by phlax** [comment]
its causing ci to fail - its not immediately clear why

my first thought was that it might be lack of `export` but im not so sure looking at the source

this is the error https://dev.azure.com/cncf/envoy/_build/results?buildId=70001&view=logs&j=e969334a-0e55-5c18-ac96-8b546753391e&t=b5329a9e-93c7-5412-9f4c-6f8d659e1b57&l=16454

if ci continues to fail - perhaps put an `pwd`  in `ci/flaky_test/run_process_xml.sh`  on line 5 just before the fail

File: ci/do_ci.sh
Comment (phlax) [inline ci/do_ci.sh:16]: im tempted to think that this is the cause of the ci fail - just on the basis of what has changed - but i dont see an obv reason to support that

**Review by phlax — APPROVED** [review]
great, thanks @murray-stripe 

these scripts are a bit of a minefield so thanks again!

**Comment by murray-stripe** [comment]
@lizan - Just checking in to see if this can be merged or if there is anything else you need from me.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**
